test/e2e/app-dir/metadata/app/opengraph/invalid-image/page.tsx

```diff
@@ -0,0 +1,9 @@
+export default function Page() {
+  return 'opengraph-article'
+}
+
+export const metadata = {
+  openGraph: {
+    images: [{ url: undefined }],
+  },
+}
```

Comment:
Instead of silently filtering it out - should we instead throw an error?

Seems like url undefined would be user error and we should fail fast so it can be corrected.

Reply:
Mentioned in the PR description

> Add filtering for now as the erroring doesn't show the proper trace pointing to where it's original introduced, might introduce other validation in the future.

Was intended to do that bu realize the error trace will be thrown from next.js that hard for users to determine where exactly it from, so we only filter for now and revisit the erroring/checking in the future

Reply:
Oh I see in the description now:

> Add filtering for now as the erroring doesn't show the proper trace pointing to where it's original introduced, might introduce other validation in the future.

I still think we could throw an error that explains something like:

```js
throw new Error('openGraph.images[0].url is undefined')
```

Reply:
The metadata could be merged among the layouts and pages so that won't actually know where it's from. and the utils is shared between `twitter` and `openGraph` resolvers now. or it could be `[null]` where we need to add few more error message branches for the helper

Reply:
Oh bummer. I guess this is an okay workaround for now.